### PR TITLE
updated readme to include new modes

### PR DESCRIPTION
**Pull request by kroos47**
- Added readme for new modes.

### CODE REVIEW

**Review by prabal-banerjee — APPROVED** [review]

**Comment by prabal-banerjee** [comment]
@DannyS03 These are the changes we want to make. Should we proceed with the merge and then you rebase your PR?

**Comment by DannyS03** [comment]
> @DannyS03 These are the changes we want to make. Should we proceed with the merge and then you rebase your PR?

Sounds good

File: README.md [hunk 1]
```diff
@@ -8,11 +8,16 @@ Light client for Data Availability Blockchain of Polygon 💻
 
 ## Introduction
 
-Naive approach for building one DA light client, which will do following
+Naive approach for building one AVAIL light client, which will do following
 
 - Listen for newly mined blocks
-- As soon as new block is available, attempts to eventually gain confidence by asking for proof from full client _( via JSON RPC interface )_ for `N` many cells where cell is defined as `{row, col}` pair
-- For lower numbered blocks, for which no confidence is yet gained, does batch processing in reverse order i.e. prioritizing latest blocks over older ones
+- As soon as new block is available, attempts to eventually gain confidence by asking for proof from full client _( via JSON RPC interface )_ for `N` many cells where cell is defined as `{row, col}` pair. The number of cells are randomly sampled to fetch the confidence eventually.
```
[the remark this answers is not in the document]
Comment (kroos47) [reply README.md:14]: yeh I will update that also

**Review by aterentic-ethernal — APPROVED** [review]

**Review by prabal-banerjee — APPROVED** [review]

File: README.md [hunk 2]
```diff
@@ -31,9 +36,13 @@ ipfs_seed = 1
 ipfs_port = 37000
 ipfs_path = "avail_ipfs_store"
 
+# put full_node_rpc = https://devnet-avail.polygon.technology/ incase you are connecting to devnet
 full_node_rpc = "http://127.0.0.1:9933"
```
Comment (prabal-banerjee) [inline README.md:40]: The config should take an array, right? Same for `ws`? @kroos47

**Review by prabal-banerjee — APPROVED** [review]

File: README.md [hunk 3]
```diff
@@ -13,12 +13,17 @@
 
 `avail-light` is a data availability light client that can do the following:
 
-* Listen for newly produced blocks.
+* Listen for newly mined blocks
```
Comment (prabal-banerjee) [inline README.md:16]: I would prefer `forged` over `mined`. However, I think right now we listen to `finalized` blocks, right?

File: README.md [hunk 4]
```diff
@@ -13,12 +13,17 @@
...
 * Gain confidence for `N` *cells*, where *cell* is defined as a `{row, col}` pair.
-  > As soon as a new block is available, the light client attempts to gain confidence by asking for a proof from a full 
-  client via JSON-RPC.
-* Batch processces blocks in reverse order (i.e. prioritizes the latest blocks) for lower numbered blocks where no confidence 
-  is yet gained.
+  > As soon as a new block is available, the light client attempts to gain confidence by asking for a proof from a full client via JSON-RPC.
+* Batch processces blocks in reverse order (i.e. prioritizes the latest blocks) for lower numbered blocks 	where no confidence is yet gained.
```
Comment (prabal-banerjee) [inline README.md:19]: This is currently happening on v1?

File: README.md [hunk 5]
```diff
@@ -13,12 +13,17 @@
...
+
+### Modes of Operation
+
+1. **Light client**: This is the basic mode of operation and is always active in whichever mode it works. If the APP_ID is not given then this mode will be commence. In this mode, the client on each header it receieves will do the random sampling using RPC calls. It gets random cells in return, which verifies and calculate the confidence.
```
Comment (prabal-banerjee) [inline README.md:23]: `will commence`
`gets proofs in return`
Some more proofreading required. Docs team can help here.

File: README.md [hunk 6]
```diff
@@ -13,12 +13,17 @@
...
+2. **App-Specific-Mode**: In this mode if an App_ID(greater than 0) is given in the config file then the client would find out the cols related the App_ID given using the app_data_lookup in the header. Then it fetches the 50% of cells from the cols, verifies them and uses them to decode the app_extrinsics_data.
```
Comment (prabal-banerjee) [inline README.md:24]: Does the decoding work in v1?

File: README.md [hunk 7]
```diff
@@ -13,12 +13,17 @@
...
+* Batch processces blocks in reverse order (i.e. prioritizes the latest blocks) for lower numbered blocks 	where no confidence is yet gained.
+
+### Modes of Operation
+
+1. **Light client**: This is the basic mode of operation and is always active in whichever mode it works. If the APP_ID is not given then this mode will be commence. In this mode, the client on each header it receieves will do the random sampling using RPC calls. It gets random cells in return, which verifies and calculate the confidence.
+2. **App-Specific-Mode**: In this mode if an App_ID(greater than 0) is given in the config file then the client would find out the cols related the App_ID given using the app_data_lookup in the header. Then it fetches the 50% of cells from the cols, verifies them and uses them to decode the app_extrinsics_data. 
+3. **Fat-Client-Mode**: In this mode, the client gets the entire extended matrix using IPFS if it is available or uses RPC calls to fetch. It verifies the entire cells and computes the CID mapping for the IPFS Pinning. It then decodes the entire matrix and reconstruct the entire app_specific_data related to all App_IDs.
```
Comment (prabal-banerjee) [inline README.md:25]: `entire cells` -> `all the cells` or `entire matrix`? 
Decodes matrix -> reconstructs original matrix or decodes extended matrix?

**Review by prabal-banerjee — CHANGES_REQUESTED** [review]
@kroos47 Suggested few changes. 

@DannyS03 IMO, more proofreading is needed (on the added content). Kindly take a pass.

File: README.md (hunk 4 above)
Comment (kroos47) [reply README.md:19]: oh rebasing did a mistake 😓  will do the correction

File: README.md (hunk 6 above)
Comment (kroos47) [reply README.md:24]: yes!! reconstruted data is given in both app and fat mode

File: README.md (hunk 7 above)
Comment (kroos47) [reply README.md:25]: I think its original matrix

Comment (kroos47) [reply README.md]: @aterentic-ethernal

File: README.md [hunk 8]
```diff
@@ -13,12 +13,16 @@
...
-  > As soon as a new block is available, the light client attempts to gain confidence by asking for a proof from a full 
-  client via JSON-RPC.
-* Batch processces blocks in reverse order (i.e. prioritizes the latest blocks) for lower numbered blocks where no confidence 
-  is yet gained.
+  > As soon as a new block is available, the light client attempts to gain confidence by asking for a proof from a full client via JSON-RPC.
+
+### Modes of Operation
+
+1. **Light client**: This is the basic mode of operation and is always active in whichever mode it works. If the APP_ID is not given then this mode will commence. In this mode, the client on each header it receieves will do the random sampling using RPC calls. It gets random cells with proofs in return, which verifies and calculate the confidence.
+2. **App-Specific-Mode**: In this mode if an App_ID(greater than 0) is given in the config file then the client would find out the cols related the App_ID given using the app_data_lookup in the header. Then it fetches the 50% of cells from the cols, verifies them and uses them to decode and reconstruct the app_extrinsics_data.
```
Comment (DannyS03) [inline README.md:23]:
```suggestion
2. **App-Specific Mode**: If an **`App_ID` > 0** is given in the config file, the client finds out the `cols` related to the provided `App_ID` using `app_data_lookup` in the header. The client then fetches 50% of cells from `cols`, verifies them, and uses them to decode and reconstruct the `app_extrinsics_data`. 
```

File: README.md [hunk 9]
```diff
@@ -13,12 +13,16 @@
...
 * Gain confidence for `N` *cells*, where *cell* is defined as a `{row, col}` pair.
-  > As soon as a new block is available, the light client attempts to gain confidence by asking for a proof from a full 
-  client via JSON-RPC.
-* Batch processces blocks in reverse order (i.e. prioritizes the latest blocks) for lower numbered blocks where no confidence 
-  is yet gained.
+  > As soon as a new block is available, the light client attempts to gain confidence by asking for a proof from a full client via JSON-RPC.
+
+### Modes of Operation
+
+1. **Light client**: This is the basic mode of operation and is always active in whichever mode it works. If the APP_ID is not given then this mode will commence. In this mode, the client on each header it receieves will do the random sampling using RPC calls. It gets random cells with proofs in return, which verifies and calculate the confidence.
```
Comment (DannyS03) [inline README.md:22]:
```suggestion
1. **Light Client Mode**: The basic mode of operation and is always active in whichever mode is operational. If an `App_ID` is not provided, this mode will commence. The client on each header it receives will do random sampling using RPC calls. It gets random cells with proofs in return, which verifies and calculates the confidence.
```

File: README.md [hunk 10]
```diff
@@ -13,12 +13,16 @@
...
+2. **App-Specific-Mode**: In this mode if an App_ID(greater than 0) is given in the config file then the client would find out the cols related the App_ID given using the app_data_lookup in the header. Then it fetches the 50% of cells from the cols, verifies them and uses them to decode and reconstruct the app_extrinsics_data. 
+3. **Fat-Client-Mode**: In this mode, the client gets the entire extended matrix using IPFS if it is available or uses RPC calls to fetch. It verifies all the cells and computes the CID mapping for the IPFS Pinning. It then decodes the extended matrix and reconstruct the app_specific_data related to all App_IDs.
```
Comment (DannyS03) [inline README.md:24]:
```suggestion
3. **Fat-Client Mode**: The client retrieves the entire extended matrix using IPFS (if available) or fetches via RPC calls. It verifies all the cells and computes the CID mapping for the IPFS Pinning. It then decodes the extended matrix and reconstructs the `app_specific_data` related to all `App_IDs`.
```

**Review by DannyS03 — CHANGES_REQUESTED** [review]

**Comment by prabal-banerjee** [comment]
@kroos47 If you can kindly check and commit. 
@vthunder Would be great if you can go through once.

File: README.md [hunk 11]
```diff
@@ -8,11 +8,16 @@ Light client for Data Availability Blockchain of Polygon 💻
...
-- For lower numbered blocks, for which no confidence is yet gained, does batch processing in reverse order i.e. prioritizing latest blocks over older ones
+- As soon as new block is available, attempts to eventually gain confidence by asking for proof from full client _( via JSON RPC interface )_ for `N` many cells where cell is defined as `{row, col}` pair. The number of cells are randomly sampled to fetch the confidence eventually.
+
+### Modes of Operation
+
+1. **Light client**: The basic mode of operation and is always active in whichever mode is operational. If an `App_ID` is not provided, this mode will commence. The client on each header it receives will do random sampling using RPC calls. It gets random cells with proofs in return, which verifies and calculates the confidence.
+2. **App-Specific-Mode**: If an **`App_ID` > 0** is given in the config file, the client finds out the `cols` related to the provided `App_ID` using `app_data_lookup` in the header. The client then fetches 50% of cells from `cols`, verifies them, and uses them to decode and reconstruct the `app_extrinsics_data`.
```
Comment (DannyS03) [inline README.md:19]:
```suggestion
2. **App-Specific Mode**: If an **`App_ID` > 0** is given in the config file, the client finds out the `cols` related to the provided `App_ID` using `app_data_lookup` in the header. The client then fetches 50% of cells from `cols`, verifies them, and uses them to decode and reconstruct the `app_extrinsics_data`. 
```

File: README.md [hunk 12]
```diff
@@ -8,11 +8,16 @@ Light client for Data Availability Blockchain of Polygon 💻
...
+
+### Modes of Operation
+
+1. **Light client**: The basic mode of operation and is always active in whichever mode is operational. If an `App_ID` is not provided, this mode will commence. The client on each header it receives will do random sampling using RPC calls. It gets random cells with proofs in return, which verifies and calculates the confidence.
+2. **App-Specific-Mode**: If an **`App_ID` > 0** is given in the config file, the client finds out the `cols` related to the provided `App_ID` using `app_data_lookup` in the header. The client then fetches 50% of cells from `cols`, verifies them, and uses them to decode and reconstruct the `app_extrinsics_data`. 
+3. **Fat-Client-Mode**: The client retrieves the entire extended matrix using IPFS (if available) or fetches via RPC calls. It verifies all the cells and computes the CID mapping for the IPFS Pinning. It then decodes the extended matrix and reconstructs the `app_specific_data` related to all `App_IDs`.
```
Comment (DannyS03) [inline README.md:20]:
```suggestion
3. **Fat-Client Mode**: The client retrieves the entire extended matrix using IPFS (if available) or fetches via RPC calls. It verifies all the cells and computes the CID mapping for the IPFS Pinning. It then decodes the extended matrix and reconstructs the `app_specific_data` related to all `App_IDs`.
```

File: README.md [hunk 13]
```diff
@@ -8,11 +8,16 @@ Light client for Data Availability Blockchain of Polygon 💻
...
+Naive approach for building one AVAIL light client, which will do following
 
 - Listen for newly mined blocks
-- As soon as new block is available, attempts to eventually gain confidence by asking for proof from full client _( via JSON RPC interface )_ for `N` many cells where cell is defined as `{row, col}` pair
-- For lower numbered blocks, for which no confidence is yet gained, does batch processing in reverse order i.e. prioritizing latest blocks over older ones
+- As soon as new block is available, attempts to eventually gain confidence by asking for proof from full client _( via JSON RPC interface )_ for `N` many cells where cell is defined as `{row, col}` pair. The number of cells are randomly sampled to fetch the confidence eventually.
+
+### Modes of Operation
+
+1. **Light client**: The basic mode of operation and is always active in whichever mode is operational. If an `App_ID` is not provided, this mode will commence. The client on each header it receives will do random sampling using RPC calls. It gets random cells with proofs in return, which verifies and calculates the confidence.
```
Comment (DannyS03) [inline README.md:18]:
```suggestion
1. **Light Client Mode**: The basic mode of operation and is always active in whichever mode is operational. If an `App_ID` is not provided, this mode will commence. The client on each header it receives will do random sampling using RPC calls. It gets random cells with proofs in return, which verifies and calculates the confidence.
```

**Review by DannyS03 — APPROVED** [review]

File: README.md [hunk 14]
```diff
@@ -11,14 +11,16 @@
 
 ## Introduction
 
-`avail-light` is a data availability light client that can do the following:
+Naive approach for building one AVAIL light client, which will do following
```
Comment (prabal-banerjee) [inline README.md:14]: @kroos47 I think the doc team changed the wordings here. Kindly keep the changes proposed by them.

File: README.md [hunk 15]
```diff
@@ -11,14 +11,16 @@
...
 
-* Listen for newly produced blocks.
-* Gain confidence for `N` *cells*, where *cell* is defined as a `{row, col}` pair.
-  > As soon as a new block is available, the light client attempts to gain confidence by asking for a proof from a full 
-  client via JSON-RPC.
-* Batch processces blocks in reverse order (i.e. prioritizes the latest blocks) for lower numbered blocks where no confidence 
-  is yet gained.
+- Listen for newly mined blocks
+- As soon as new block is available, attempts to eventually gain confidence by asking for proof from full client _( via JSON RPC interface )_ for `N` many cells where cell is defined as `{row, col}` pair. The number of cells are randomly sampled to fetch the confidence eventually.
```
Comment (prabal-banerjee) [inline README.md:17]: Finalized block, right?

File: README.md [hunk 16]
```diff
@@ -11,14 +11,16 @@
...
+
+### Modes of Operation
+
+1. **Light-client Mode**: The basic mode of operation and is always active in whichever mode is operational. If an `App_ID` is not provided, this mode will commence. The client on each header it receives will do random sampling using RPC calls. It gets random cells with proofs in return, which verifies and calculates the confidence.
+2. **App-Specific Mode**: If an **`App_ID` > 0** is given in the config file, the client finds out the `cols` related to the provided `App_ID` using `app_data_lookup` in the header. The client then fetches 50% of cells from `cols`, verifies them, and uses them to decode and reconstruct the `app_extrinsics_data`.
```
Comment (prabal-banerjee) [inline README.md:22]: Maybe we can simplify and say that the relevant cells are downloaded and the data is reconstructed.

File: README.md [hunk 17]
```diff
@@ -11,14 +11,16 @@
...
-  > As soon as a new block is available, the light client attempts to gain confidence by asking for a proof from a full 
-  client via JSON-RPC.
-* Batch processces blocks in reverse order (i.e. prioritizes the latest blocks) for lower numbered blocks where no confidence 
-  is yet gained.
+- Listen for newly mined blocks
+- As soon as new block is available, attempts to eventually gain confidence by asking for proof from full client _( via JSON RPC interface )_ for `N` many cells where cell is defined as `{row, col}` pair. The number of cells are randomly sampled to fetch the confidence eventually.
+
+### Modes of Operation
+
+1. **Light-client Mode**: The basic mode of operation and is always active in whichever mode is operational. If an `App_ID` is not provided, this mode will commence. The client on each header it receives will do random sampling using RPC calls. It gets random cells with proofs in return, which verifies and calculates the confidence.
```
Comment (prabal-banerjee) [inline README.md:21]: `gets random cells with proofs` -> It gets the cells it asked for.

**Review by prabal-banerjee — COMMENTED** [review]
Need to confirm whether all the changes proposed by doc team is there on the PR.

**Review by prabal-banerjee — APPROVED** [review]

**Review by DannyS03 — APPROVED** [review]

**Review by aterentic-ethernal — APPROVED** [review]

**Pull request merged**